jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +697,299 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
+            'xor', 'andnot']
+
+        max_num_regions : int or `None`
+            Maximum number of regions to load, starting from top of the list.
+            Default is to load everything.
+
+        refdata_label : str or `None`
+            Label of data to use for sky-to-pixel conversion for a region, or
+            mask creation. Data must already be loaded into Jdaviz.
+            If `None`, defaults to the reference data in the default viewer.
+            Choice of this data is particularly important when sky
+            region is involved.
+
+        return_bad_regions : bool
+            If `True`, return the regions that failed to load (see ``bad_regions``);
+            This is useful for debugging. If `False`, do not return anything (`None`).
+
+        kwargs : dict
+            Extra keywords to be passed into the region's ``to_mask`` method.
+            **This is ignored if the region can be made interactive.**
+
+        Returns
+        -------
+        bad_regions : list of (obj, str) or `None`
+            If requested (see ``return_bad_regions`` option), return a
+            list of ``(region, reason)`` tuples for region objects that failed to load.
+            If all the regions loaded successfully, this list will be empty.
+            If not requested, return `None`.
+
+        """
+        if len(self.app.data_collection) == 0:
+            raise ValueError('Cannot load regions without data.')
+
+        from photutils.aperture import (CircularAperture, SkyCircularAperture,
+                                        EllipticalAperture, SkyEllipticalAperture,
+                                        RectangularAperture, SkyRectangularAperture,
+                                        CircularAnnulus, SkyCircularAnnulus)
+        from regions import (Regions, CirclePixelRegion, CircleSkyRegion,
+                             EllipsePixelRegion, EllipseSkyRegion,
+                             RectanglePixelRegion, RectangleSkyRegion,
+                             CircleAnnulusPixelRegion, CircleAnnulusSkyRegion)
+        from jdaviz.core.region_translators import regions2roi, aperture2regions
+
+        if not isinstance(regions, (list, tuple, Regions, SpectralRegion)):
+            regions = [regions]
+
+        n_loaded = 0
+        bad_regions = []
+
+        # To keep track of masked subsets.
+        msg_prefix = 'MaskedSubset'
+        msg_count = _next_subset_num(msg_prefix, self.app.data_collection.subset_groups)
+
+        viewer_parameter = kwargs.pop('viewer', None)
+        viewer_name = viewer_parameter or list(self.app._jdaviz_helper.viewers.keys())[0]
+        viewer = self.app.get_viewer(viewer_name)
+        # Subset is global but reference data is viewer-dependent.
+        if refdata_label is None:
+            data = viewer.state.reference_data
+        else:
+            data = self.app.data_collection[refdata_label]
+
+        # TODO: Make this work for data cube.
```

Comment:
is this a new TODO comment or copied from the previous region import code?

Reply:
Copied from previous, I'll remove it.

Reply:
does it work for cubes?  If not, let's make sure we have a ticket (but safe to ignore here if this was from moved code)

Reply:
Seems like this works, although the solution is hacky (per https://github.com/glue-viz/glue-astronomy/issues/75 ), maybe @pllim can clarify on the status. Subset creation works with cubes so I think this is safe to remove but I'll wait for Pey-Lian's answer.

Reply:
@rosteen reworked some spectrum WCS stuff recently so I'll defer to him. Or was it @dhomeier ? As long as the tests don't fail and results don't change, whatever works is fine by me.

Reply:
If it works it works 🤷 I don't recall changing the spectrum WCS stuff recently in a way that would affect this.

Reply:
That issue has been closed by https://github.com/glue-viz/glue-astronomy/issues/75#issuecomment-2359390639, and the translator would only use a `PaddedSpectrumWCS` if the cube had a 1D WCS, which I assume is not the case here.